src/Workspaces/Core/Portable/Rename/ConflictResoution.cs

```diff
@@ -28,22 +32,43 @@ namespace Microsoft.CodeAnalysis.Rename
         private readonly ImmutableDictionary<DocumentId, ImmutableArray<RelatedLocation>> _documentToRelatedLocationsMap;
 
         public ConflictResolution(
-            Solution oldSolution, Solution newSolution, bool replacementTextValid,
+            Solution oldSolution,
+            Solution newSolutionWithoutRenamedDocument,
+            bool replacementTextValid,
+            DocumentId renamedDocumentId,
+            string renamedDocumentNewName,
             ImmutableArray<DocumentId> documentIds, ImmutableArray<RelatedLocation> relatedLocations,
             ImmutableDictionary<DocumentId, ImmutableArray<(TextSpan oldSpan, TextSpan newSpan)>> documentToModifiedSpansMap,
             ImmutableDictionary<DocumentId, ImmutableArray<ComplexifiedSpan>> documentToComplexifiedSpansMap,
             ImmutableDictionary<DocumentId, ImmutableArray<RelatedLocation>> documentToRelatedLocationsMap)
         {
             OldSolution = oldSolution;
-            NewSolution = newSolution;
+            _newSolutionWithoutRenamedDocument = newSolutionWithoutRenamedDocument;
             ReplacementTextValid = replacementTextValid;
+            _renamedDocumentId = renamedDocumentId;
+            _renamedDocumentNewName = renamedDocumentNewName;
             DocumentIds = documentIds;
             RelatedLocations = relatedLocations;
             _documentToModifiedSpansMap = documentToModifiedSpansMap;
             _documentToComplexifiedSpansMap = documentToComplexifiedSpansMap;
             _documentToRelatedLocationsMap = documentToRelatedLocationsMap;
         }
 
+        /// <summary>
+        /// The final solution snapshot
+        /// </summary>
+        public Solution NewSolution
+        {
+            get
+            {
+                var newSolution = _newSolutionWithoutRenamedDocument;
+                if (_renamedDocumentId != null)
```

Comment:
Is there any reason we can't just do this in the constructor?

Reply:
yes, we could do that :)  i can do in followup.

Reply:
Done in https://github.com/dotnet/roslyn/pull/43592